crates/blockifier/src/transaction/objects.rs

```diff
@@ -103,11 +104,12 @@ pub struct ResourcesMapping(pub IndexMap<String, u64, HasherBuilder>);
 #[cfg(feature = "parity-scale-codec")]
 impl Encode for ResourcesMapping {
     fn size_hint(&self) -> usize {
-        self.0.len() * core::mem::size_of::<u64>()
+        self.0.len() * (core::mem::size_of::<String>() + core::mem::size_of::<u64>())
```

Comment:
ca non `core::mem::size_of::<String>()`

La size in memory d'une string c'est un pointer, une len et une capacity. Donc 3 * sizeof::<u64>(). La on parle de la longuque de la string en elle même. Et ça on peut pas savoir sans iterer